File: JavaServletSrc/arrt_crea_mass_cf.js
function arrt_crea_mass_cf() {
var w_tiposog;
var w_tipoinv;
var w_maxinvii;
var w_a2013;
var w_a2014;
var w_a2015;
var w_a2016;
var w_a2017;
var w_a2018;
var w_a2019;
var w_a2020;
var w_a2021;
var w_a2022;
var m_Caller
arrt_crea_mass_cf._p=_rargs(arguments,{eval:function(e){eval(e)}})
function i_BlankParameters() {
w_tiposog='';
w_tipoinv='';
w_maxinvii=0;
w_a2013='';
w_a2014='';
w_a2015='';
w_a2016='';
w_a2017='';
w_a2018='';
w_a2019='';
w_a2020='';
w_a2021='';
w_a2022='';
}
if(arrt_crea_mass_cf._p.length>0)_rargs([[['w_tiposog',arrt_crea_mass_cf._p[0]],['w_tipoinv',arrt_crea_mass_cf._p[1]],['w_maxinvii',arrt_crea_mass_cf._p[2]],['w_a2013',arrt_crea_mass_cf._p[3]],['w_a2014',arrt_crea_mass_cf._p[4]],['w_a2015',arrt_crea_mass_cf._p[5]],['w_a2016',arrt_crea_mass_cf._p[6]],['w_a2017',arrt_crea_mass_cf._p[7]],['w_a2018',arrt_crea_mass_cf._p[8]],['w_a2019',arrt_crea_mass_cf._p[9]],['w_a2020',arrt_crea_mass_cf._p[10]],['w_a2021',arrt_crea_mass_cf._p[11]],['w_a2022',arrt_crea_mass_cf._p[12]]]],{eval:function(e){eval(e)}})
arrt_crea_mass_cf._s=new BatchJavascript();
if (Ne(typeof(document.FSender),'undefined') && Ne(typeof(document.FSender.m_cRegionalSettings),'undefined')) {
arrt_crea_mass_cf._s.SetParameterString('m_cRegionalSettings',document.FSender.m_cRegionalSettings.value);
}
arrt_crea_mass_cf._s.SetParameterString('w_tiposog',WtA(w_tiposog,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_tipoinv',WtA(w_tipoinv,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_maxinvii',WtA(w_maxinvii,'N'));
arrt_crea_mass_cf._s.SetParameterString('w_a2013',WtA(w_a2013,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2014',WtA(w_a2014,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2015',WtA(w_a2015,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2016',WtA(w_a2016,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2017',WtA(w_a2017,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2018',WtA(w_a2018,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2019',WtA(w_a2019,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2020',WtA(w_a2020,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2021',WtA(w_a2021,'C'));
arrt_crea_mass_cf._s.SetParameterString('w_a2022',WtA(w_a2022,'C'));
arrt_crea_mass_cf._s.SetParameterString('m_bApplet','true');
arrt_crea_mass_cf._s.CallServlet('arrt_crea_mass_cf');
if (Eq(typeof(m_bCalculating),'undefined') ||  ! (m_bCalculating)) {
m_cErrorFromRoutine=arrt_crea_mass_cf._s.errmsg;
}
return arrt_crea_mass_cf._s.GetString();
}
arrt_crea_mass_cf.CtxLoad_ = function(){return []}

File: JavaServletSrc/arrt_crea_mass_cf.java
import java.io.*;
import java.text.*;
import java.util.*;
import javax.servlet.*;
import javax.servlet.http.*;
import com.zucchetti.sitepainter.SPPrxycizer;

@com.zucchetti.sitepainter.EntityRoles({})
public class arrt_crea_mass_cf extends SPRoutineServlet implements SPInvokable {
  public static String m_cEntityUid = "!*5cf17532*!";
  public class ServletStatus extends SPRoutineServletStatus {
    public String gMsgImp;
    public String gMsgProc;
    public String gAzienda;
    public String gIntemediario;
    public String gPathDoc;
    public String fhand;
    public String _nomefile;
    public String cNomeFile;
    public String _mese;
    public String _riga;
    public String _filtroanni;
    public String w_Orderby;
    public String _blocco;
    public double _contarec;
    public double _tipo1;
    public double _tipo2;
    public double _max;
    public double _min;
    public double _tot;
    public double _mono;
    public String _oldmsg;
    public double _reccount;
    public double _cicli;
    public double _contacicli;
    public double _bottom;
    public double _top;
    public String _cfint;
    public String _rsint;
    public String _codint;
    public double _contarighe;
    public double _contat1;
    public double _contat2;
    public double _maxrig;
    public String cRitorno;
    public String _newline;
    public String w_tiposog;
    public String w_tipoinv;
    public double w_maxinvii;
    public String w_a2013;
    public String w_a2014;
    public String w_a2015;
    public String w_a2016;
    public String w_a2017;
    public String w_a2018;
    public String w_a2019;
    public String w_a2020;
    public String w_a2021;
    public String w_a2022;
    public String aaabaaaaaaaaa_retval;
    public void GetNonLocals(SPParameterSource source) {
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_tiposog","w_tiposog");
      }        
      w_tiposog = source.GetParameter("w_tiposog","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_tipoinv","w_tipoinv");
      }        
      w_tipoinv = source.GetParameter("w_tipoinv","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_maxinvii","w_maxinvii");
      }        
      w_maxinvii = source.GetParameter("w_maxinvii",0);
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2013","w_a2013");
      }        
      w_a2013 = source.GetParameter("w_a2013","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2014","w_a2014");
      }        
      w_a2014 = source.GetParameter("w_a2014","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2015","w_a2015");
      }        
      w_a2015 = source.GetParameter("w_a2015","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2016","w_a2016");
      }        
      w_a2016 = source.GetParameter("w_a2016","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2017","w_a2017");
      }        
      w_a2017 = source.GetParameter("w_a2017","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2018","w_a2018");
      }        
      w_a2018 = source.GetParameter("w_a2018","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2019","w_a2019");
      }        
      w_a2019 = source.GetParameter("w_a2019","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2020","w_a2020");
      }        
      w_a2020 = source.GetParameter("w_a2020","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2021","w_a2021");
      }        
      w_a2021 = source.GetParameter("w_a2021","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_a2022","w_a2022");
      }        
      w_a2022 = source.GetParameter("w_a2022","");
    }
  }
  public SPEntityGlobals getEntityGlobals() {
    // Costanti globali dell' entita'
    SPRoutineGlobals l_eg;
    l_eg = new SPRoutineGlobals();
    l_eg.name = "arrt_crea_mass_cf";
    l_eg.hasProgressItem = false;
    l_eg.isPublic = false;
    return l_eg;
  }
  public void doProcess(HttpServletRequest request,HttpServletResponse response) throws IOException, ServletException {
    // Inizializzazione delle variabili della classe globale di stato
    ServletStatus status;
    status = new ServletStatus();
    status.containing = this;
    SPParameterSource source;
    source = SPLib.GetSource(request);
    if ( ! (EnterServlet(status,source,request,response))) {
      return;
    }
    Begin(status);
  }
  boolean NeedsFetchPage(ServletStatus status) {
    return false;
  }
  private void ExecuteRoutine(ServletStatus status) throws IOException {
    arrt_crea_mass_cfR Rcalled;
    Rcalled = new arrt_crea_mass_cfR(status.context,status);
    if (CPLib.ne(status.m_cAction,"executeAsync")) {
      ExecuteWorkflowStartingScript(status,Rcalled);
      status.aaabaaaaaaaaa_retval = Rcalled.Run(status.w_tiposog,status.w_tipoinv,status.w_maxinvii,status.w_a2013,status.w_a2014,status.w_a2015,status.w_a2016,status.w_a2017,status.w_a2018,status.w_a2019,status.w_a2020,status.w_a2021,status.w_a2022);
      if (Rcalled.m_bError) {
        status.m_bError = true;
        status.m_cErrMsg = Rcalled.m_cLastMsgError;
      } else if ( ! (CPLib.Empty(status.m_cFaultTrace))) {
        status.m_bError = true;
        status.m_cErrMsg = status.m_cFaultTrace;
      }
      if (CPLib.eqr(CPLib.Left(status.m_cAtExit,8),"workflow")) {
        ExitToWorkflow(status);
      }
      if (Forward.IsForwarded(status.forwardedto)) {
        if (CPLib.IsNull(status.forwardedto.GetParameterString("m_cAtExit",null))) {
          status.forwardedto.SetParameter("m_cAtExit",status.m_cAtExit);
        }
        SPLib.ForwardTo(status.forwardedto,servletContext,status.request,status.response);
        return;
      }
    }
    BringBackPage(status,Rcalled);
  }
  CallerWithObjs NewRoutine(SPServletStatus status,Caller dataFromClient) {
    return new arrt_crea_mass_cfR(status.context,dataFromClient);
  }
  protected void BringBackPage(ServletStatus status,arrt_crea_mass_cfR Rcalled) throws IOException {
    String l_cErrMsg;
    String msg;
    msg = status.context.Translate("MSG_END_ROUTINE");
    SPLib.SetContentType(status.response);
    SPLib.NoCache(status.response);
    status.out = status.response.getWriter();
    String l_cJsUid;
    status.out.println(
     "<!doctype html>" +
    "");
    status.out.println(
     "<html>" +
    "");
    status.out.println(
     "<head>" +
    "");
    status.out.println(
     "<title>" +
     ""+status.context.Translate("MSG_ROUTINE_WIND")+"" +
     "</title>" +
    "");
    l_cJsUid = SPPrxycizer.proxycizedPath("stdFunctions.js");
    status.out.println(
     "<script src='../"+l_cJsUid+"' type='text/javascript'>" +
     "</script>" +
    "");
    l_cJsUid = SPPrxycizer.proxycizedPath("controls.js");
    status.out.println(
     "<script src='../"+l_cJsUid+"' type='text/javascript'>" +
     "</script>" +
    "");
    status.out.println(
     "<link type='text/css' rel='STYLESHEET' href='../"+status.m_cTheme+"/formPage.css'>" +
    "");
    status.out.println(
     "</head>" +
    "");
    if (status.m_bError) {
      msg = status.context.Translate("MSG_END_ROUTINE_WITH_MSG");
      status.out.println(
       "<body onkeydown='window.ManageKeys(event,false,false)'>" +
      "");
    } else {
      status.out.println(
       "<body>" +
      "");
    }
    status.out.println(
     "<p class='TabLabel' style='margin:0'>" +
     ""+msg+"" +
     "</p>" +
    "");
    if (status.m_bError) {
      status.out.println(
       "<script type='text/javascript'>" +
      "");
      l_cErrMsg = status.context.Translate(com.zucchetti.sitepainter.Library.explicativeErrorMessage(status.m_cErrMsg));
      status.out.println("detailed_database_error="+SPLib.ToJSValue(l_cErrMsg,"C",0,0)+";");
      status.out.println("m_cBrowser=" + SPLib.ToJSValue(status.m_cBrowser,"C",0,0)+ ";");
      status.out.println(
       "</script>" +
      "");
    }
    status.out.println(
     "<!--" +
    "");
    if ( ! (CPLib.Empty(Rcalled.m_cLastMsgError))) {
      CPTokenSource ts;
      ts = new CPTokenSource(status.context.Translate(com.zucchetti.sitepainter.Library.explicativeErrorMessage(Rcalled.m_cLastMsgError)),"\n");
      while (ts.HasMoreTokens()) {
        status.out.println("Error message:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(ts.NextToken())+"");
      }
    }
    if ( ! (CPLib.Empty(status.m_cFaultTrace))) {
      status.out.println("Fault:["+SPLib.ToJSValue(status.context.Translate("MSG_ROUTINE_FAULTED","arrt_crea_mass_cf"),"C",0,0)+","+SPLib.ToJSValue(status.m_cFaultTrace,"C",0,0)+"]");
      if (CPLib.Empty(Rcalled.m_cLastMsgError) &&  ! (status.m_bApplet)) {
        status.out.println("Error message:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(status.m_cFaultTrace)+"");
      }
    }
    status.PrintMutableParameters();
    status.out.println("Function return value:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(CPLib.ToProperties(status.aaabaaaaaaaaa_retval))+"");
    status.out.println(
     " -->" +
    "");
    status.out.println(
     "</body>" +
    "");
    status.out.println(
     "</html>" +
    "");
  }
  private void Begin(ServletStatus status) throws IOException {
    if ( ! (CPLib.Empty(status.m_cGlobalsToRead))) {
      JSGlobals(status);
    } else if (CPLib.Empty(status.m_cBrowseName)) {
      ExecuteRoutine(status);
    } else {
      JSResultSet(status);
    }
  }
  void IncludeJS(ServletStatus status) {
    /* --- Area Manuale = UI - Include */
    /* --- Fine Area Manuale */
  }
  // m_CallableNames inizializzato dal primo accesso a GetCallableNames
  private static volatile String[] m_CallableNames;
  public static String[] GetCallableNames() {
    if (CPLib.IsNull(m_CallableNames)) {
      java.util.Set building = new java.util.HashSet();
      GetCallableNames(building);
      m_CallableNames = CPLib.SetOfStringsAsArray(building);
    }
    return (String[])m_CallableNames.clone();
  }
  public static void GetCallableNames(java.util.Set callable) {
  }
  String[] RoutineIDChain() {
    return GetCallableNames();
  }
}
